### PR TITLE
Release v0.5

### PR DESCRIPTION
**Pull request by giovannipizzi**
- Various general (but very important) speed improvements
- Adding callbacks to a number of functions (e.g. export, add_objects_to_pack, ...) to allow showing progress bars or similar indicators
- Implement repacking (at least when not changing hashing or compression)
- Remove `export`, implement `import_objects` function instead on the other side (more efficient)
- Add support for VACUUMing operations (very important for efficiency)
- Added support for multiple hashing algorithms
- Added concept of (unique) container_id
- Generalized the compression algorithm, multiple algorithms supported now

### CODE REVIEW

**Comment by giovannipizzi** [comment]
Note: once it's merged in master, I will tag it, release it on pypi and merge back in develop

**Review by chrisjsewell — CHANGES_REQUESTED** [review]
Should this not contain an update to the CHANGELOG?

Also is there a reason for this to not be a merge of develop into master, rather than a personal fork?
TBH I think this project is too small for the whole master/develop setup; it just overcomplicates things. But I'm not going to die on that hill 😉

**Comment by chrisjsewell** [comment]
You might also want to consider integrating the PyPi release into GitHub actions:

```yaml
  publish:

    name: Publish to PyPi
    needs: [pre-commit, tests]
    if: github.event_name == 'push' && startsWith(github.event.ref, 'refs/tags')
    runs-on: ubuntu-latest
    steps:
      - name: Checkout source
        uses: actions/checkout@v2
      - name: Set up Python 3.7
        uses: actions/setup-python@v1
        with:
          python-version: 3.7
      - name: Build package
        run: |
          pip install wheel
          python setup.py sdist bdist_wheel
      - name: Publish
        uses: pypa/gh-action-pypi-publish@v1.1.0
        with:
          user: __token__
          password: ${{ secrets.PYPI_KEY }}
```

**Comment by giovannipizzi** [comment]
I finally got to add the changelog. I think from now on it's enough to just mention the month (the exact date is anyway in the git history). Ready to be merged for me, please let me know when it's done so I release (thanks @chrisjsewell for the hints - I think we can improve the release cycle in the future, maybe you're right that it's more complex than needed, I'm just used to it so I try to do the same in all repos, but we can definitely improve for the future versions).

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,13 @@
+# v0.5.0 (November 2020)
+- Various general (but very important) speed improvements
```
Comment (sphuber) [inline CHANGELOG.md:2]: Just FYI, in the changelogs of most other projects of `aiidateam` we include a link to the relevant PR, e.g.:
- Add concept of (unique) `container_id` [[#97]](https://github.com/aiidateam/disk-objectstore/pull/97)
Up to you if you want to add it.

Comment (giovannipizzi) [reply CHANGELOG.md]: Indeed, I didn't bother doing it here, but it's a good thing. It's added now (only for v0.5, for older versions it's not really worth it I think, they weren't production-ready).
If you or @chrisjsewell approve it and merge it, let me know please so I release it on pypi.

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,13 @@
+# v0.5.0 (November 2020)
+- Various general (but very important) speed improvements [[#96]](https://github.com/aiidateam/disk-objectstore/pull/96) [[#102]](https://github.com/aiidateam/disk-objectstore/pull/102)
+- Add callbacks to a number of functions (e.g. export, add_objects_to_pack, ... to allow showing progress bars or similar indicators [[#96]](https://github.com/aiidateam/disk-objectstore/pull/96)
+- Implement repacking (at least when not changing hashing or compression) [[#96]](https://github.com/aiidateam/disk-objectstore/pull/96)
+- Remove `export` function, implement `import_objects` function instead, to be called on the other side (it's more efficient) [[#96]](https://github.com/aiidateam/disk-objectstore/pull/96)
+- Add support for VACUUMing operations on the SQLite database (very important for efficiency) [[#96]](https://github.com/aiidateam/disk-objectstore/pull/96)
```
Comment (sphuber) [inline CHANGELOG.md:6]: Just a quick check that it is intended that all of these point to the same PR. Maybe all these points were all addressed in the same. If this is OK, then it looks good for me to merge

**Review by sphuber — APPROVED** [review]

File: CHANGELOG.md (hunk 2 above)
Comment (giovannipizzi) [reply CHANGELOG.md:6]: Hi, yes. In the end I started addressing one issue, and one thing lead to another, and I ended up with a single "monster" PR - I know it's not a good idea in general but some of the features where quite interconnected.

**Pull request merged**